api/src/main/java/io/minio/messages/SelectObjectContentRequest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2019 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio.messages;
+
+import org.xmlpull.v1.XmlPullParserException;
+
+import com.google.api.client.util.Key;
+
+
+/**
+ * Helper class to generate Amazon AWS S3 request XML for SelectObjectContentRequest information.
+ */
+@edu.umd.cs.findbugs.annotations.SuppressFBWarnings(value = "URF_UNREAD_FIELD")
+public class SelectObjectContentRequest extends XmlEntity {
+  @Key("Expression")
+  private String expression;
+  @Key("ExpressionType")
+  private String expressionType;
+  @Key("RequestProgress")
+  private RequestProgress requestProgress;
+  @Key("InputSerialization")
+  private InputSerialization inputSerialization;
+  @Key("OutputSerialization")
+  private OutputSerialization outputSerialization;
+  @Key("ScanRange")
+  private ScanRange scanRange;
+
+
+  /**
+   * Constructs new SelectObjectContentRequest object for given parameters.
+   */
+  public SelectObjectContentRequest(String expression, boolean requestProgress, InputSerialization is,
+                                    OutputSerialization os, Long scanStartRange, Long scanEndRange)
+    throws XmlPullParserException {
+    super();
+    this.name = "SelectObjectContentRequest";
```

Comment:
Where is the name defined in this class ? Ran an example code and its erroring out as `java.lang.IllegalArgumentException`

Reply:
My suggestion is to explore the source code and find how XML encoding/decoding is done in minio-java. If you have any difficulties, let me know.

Please report any error with appropriate traces including your java environment detail with source code.

Reply:
java version `openjdk version "11.0.5" 2019-10-15`
The below code is from this PR `API.md` file. 
 ``` 
 try {
      /* play.min.io for test and development. */
      MinioClient minioClient = new MinioClient("https://play.min.io", "Q3AM3UQ867SPQQA43P2F", "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG");
      minioClient.traceOn(System.out);
      String sqlExpression = "select * from S3Object";
      InputSerialization is = InputSerialization.csv(null, false, null, null, FileHeaderInfo.USE, null, null, null);
      OutputSerialization os = OutputSerialization.csv(null, null, null, QuoteFields.ASNEEDED, null);
      SelectResponseStream stream = minioClient.selectObjectContent("sinha", "test.csv", sqlExpression, is, os, true, null, null, null);
      byte[] buf = new byte[512];
      int bytesRead = stream.read(buf, 0, buf.length);
      System.out.println(new String(buf, 0, bytesRead, StandardCharsets.UTF_8));
      Stats stats = stream.stats();
      System.out.println("bytes scanned: " + stats.bytesScanned());
      System.out.println("bytes processed: " + stats.bytesProcessed());
      System.out.println("bytes returned: " + stats.bytesReturned());
      stream.close();
    } catch (MinioException e) {
      System.out.println("Error occurred: " + e);
    }
```

Output 
```
/usr/lib/jvm/java-1.11.0-openjdk-amd64/bin/java -javaagent:/opt/idea-IC-191.6183.87/lib/idea_rt.jar=34865:/opt/idea-IC-191.6183.87/bin -Dfile.encoding=UTF-8 -classpath /home/ashish/code/go/src/github.com/minio/minio-java/api/build/libs/minio-6.0.12-DEV-all.jar:/home/ashish/client/target/classes SelectObjectExample
---------START-HTTP---------
GET /sinha?location= HTTP/1.1
Host: play.min.io
User-Agent: MinIO (amd64; amd64) minio-java/dev
x-amz-content-sha256: UNSIGNED-PAYLOAD
x-amz-date: 20191221T015201Z
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20191221/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*

HTTP/1.1 200
Server: nginx/1.10.3 (Ubuntu)
Date: Sat, 21 Dec 2019 01:52:03 GMT
Content-Type: application/xml
Content-Length: 137
Connection: keep-alive
Accept-Ranges: bytes
Content-Security-Policy: block-all-mixed-content
Vary: Origin
X-Amz-Bucket-Region: us-east-1
X-Amz-Request-Id: 15E23FF0EAC2F5B8
X-Xss-Protection: 1; mode=block

----------END-HTTP----------
Exception in thread "main" java.lang.IllegalArgumentException
	at com.google.api.client.repackaged.com.google.common.base.Preconditions.checkArgument(Preconditions.java:108)
	at com.google.api.client.util.Preconditions.checkArgument(Preconditions.java:37)
	at com.google.api.client.util.DataMap.<init>(DataMap.java:45)
	at com.google.api.client.util.Data.mapOf(Data.java:186)
	at com.google.api.client.xml.XmlNamespaceDictionary.computeAliases(XmlNamespaceDictionary.java:248)
	at com.google.api.client.xml.XmlNamespaceDictionary.computeAliases(XmlNamespaceDictionary.java:266)
	at com.google.api.client.xml.XmlNamespaceDictionary.computeAliases(XmlNamespaceDictionary.java:266)
	at com.google.api.client.xml.XmlNamespaceDictionary.computeAliases(XmlNamespaceDictionary.java:266)
	at com.google.api.client.xml.XmlNamespaceDictionary.startDoc(XmlNamespaceDictionary.java:236)
	at com.google.api.client.xml.XmlNamespaceDictionary.serialize(XmlNamespaceDictionary.java:227)
	at com.google.api.client.xml.XmlNamespaceDictionary.toStringOf(XmlNamespaceDictionary.java:174)
	at com.google.api.client.xml.GenericXml.toString(GenericXml.java:66)
	at io.minio.MinioClient.executeReq(MinioClient.java:1092)
	at io.minio.MinioClient.execute(MinioClient.java:1078)
	at io.minio.MinioClient.selectObjectContent(MinioClient.java:5818)
	at SelectObjectExample.main(SelectObjectExample.java:48)

```
Can you tell what wrong i am doing here.

Reply:
1. Java 11 is not supported yet.
2. Please pull latest code from this PR and test with java 8/9/10.